src/Core/Framework/Api/Sync/SyncService.php

```diff
@@ -144,11 +144,10 @@ private function handleCriteriaDelete(SyncOperation $operation, Context $context
     {
         $definition = $this->registry->getByEntityName($operation->getEntity());
 
-        if (!$definition instanceof MappingEntityDefinition) {
```

Comment:
@keulinho this would allow to use the criteria logic for all type of entities. Requested multiple times already.

Reply:
Sounds good 👍 

We should add the "criteria" then to the openApi spec as well: `src/Core/Framework/Api/ApiDefinition/Generator/Schema/AdminApi/paths/sync.json`

And please add a changelog with a '# Upgrade Information' section "announcing" the new feature